src/storage-controller/src/lib.rs

```diff
@@ -847,7 +847,7 @@ where
         // We cannot register tables at the table worker when in read-only mode.
         // When coming out of read-only mode (via `allow_writes()`), we will
         // register all tables that are known by that point.
-        if !table_registers.is_empty() && !self.read_only {
```

Comment:
Why did we remove the `read_only` check?

Reply:
So that you can write the initial data to the new built-in tables. That initial write goes through this code path, right?

Reply:
Maybe @aljoscha can help clarify this. My understanding is that all this code does is register a shard in the txn-wal shard. However, we don't want to register the migrated shards with txn-wal, we want to write to them directly

Reply:
Oh, I see. The legacy implementation has it's own notion of registering tables which this will now hit in read-only mode.

Reply:
Yes, exactly!